lifelines/statistics.py

```diff
@@ -180,23 +149,56 @@ def multivariate_logrank_test(event_durations, groups, event_observed=None,
     assert abs(Z_j.sum()) < 10e-8, "Sum is not zero."  # this should move to a test eventually.
 
     # compute covariance matrix
-    V_ = n_ij.mul(np.sqrt(d_i) / n_i, axis='index').fillna(1)
+    factor = (((n_i - d_i) / (n_i - 1)).replace(np.inf, 1)) * d_i
+    n_ij['_'] = n_i.values
+    V_ = n_ij.mul(np.sqrt(factor) / n_i, axis='index').fillna(1)
     V = -np.dot(V_.T, V_)
     ix = np.arange(n_groups)
-    V[ix, ix] = V[ix, ix] + ev
+    V[ix, ix] = -V[-1, ix] + V[ix, ix]
+    V = V[:-1, :-1]
 
     # take the first n-1 groups
-    U = Z_j.ix[:-1].dot(np.linalg.pinv(V[:-1, :-1]).dot(Z_j.ix[:-1]))  # Z.T*inv(V)*Z
+    U = Z_j.iloc[:-1].dot(np.linalg.pinv(V[:-1, :-1]).dot(Z_j.iloc[:-1]))  # Z.T*inv(V)*Z
 
     # compute the p-values and tests
     test_result, p_value = chisq_test(U, n_groups - 1, alpha)
-    summary = pretty_print_summary(test_result, p_value, U, t_0=t_0, test='logrank',
-                                   alpha=alpha, null_distribution='chi squared',
-                                   df=n_groups - 1, **kwargs)
 
-    if not suppress_print:
-        print(summary)
-    return summary, p_value, test_result
+    return StatisticalResult(test_result, p_value, U, t_0=t_0, test='logrank',
+                             alpha=alpha, null_distribution='chi squared',
+                             df=n_groups - 1, **kwargs)
+
+
+class StatisticalResult(object):
+
+    def __init__(self, test_results, p_value, test_statistic, **kwargs):
+        self.p_value = p_value
+        self.test_statistic = test_statistic
+        self.test_results = test_results
```

Comment:
Two questions here

1. Given the name "test_results", I am guessing that it can be a list/array/dataframe or results?
2. Should it ever be anything other than boolean(s)? If not, should it perhaps be called "significant"? Or replaced by a method like:

    @property
    def is_significant(self):
        return self.p_value <= self.alpha

?